signalfx/client.py

```diff
@@ -241,18 +272,366 @@ def _send(self):
+                                                query)
+        if order_by is not None:
+            url_to_get += '&orderBy=' + order_by
+        # for offset and limit, use API defaults (by leaving them out of url)
+        if offset is not None:
+            url_to_get += '&offset=' + str(offset)
+        if limit is not None:
+            url_to_get += '&limit=' + str(limit)
+        _timeout = timeout or self._timeout
+        resp = self._get(url_to_get, session=self._api_session,
+                         timeout=_timeout)
+        resp.raise_for_status()
+        return resp.json()
+
+    def _get_object_by_name(self, object_name, object_endpoint, timeout=None):
+        """
+        generic function to get object (metadata, tag, ) by name from SignalFx.
+
+        Args:
+            object_name (string): name of the object (e.g. 'jvm.cpu.load')
+            object_endpoint (string): API endpoint suffix (e.g. 'v2/tag')
+
+        Returns:
+            dictionary of response
+        """
+        _timeout = timeout or self._timeout
+        resp = self._get('{0}/{1}/{2}'.format(self._api_endpoint,
+                                              object_endpoint, object_name),
+                         session=self._api_session, timeout=_timeout)
+        resp.raise_for_status()
+        return resp.json()
+
+    def get_time_series_window(self, query, start_ms, end_ms, resolution=None,
+                               timeout=None):
+        """
+        gets the data within window specified by start_ms, end_ms for MTS
+            matching query
+
+        Args:
+            query (string): elasticsearch string query specifying metric time
+                series
+            start_ms (int): milliseconds since Unix epoch since which to
+                find data points
+            end_ms (int): milliseconds since Unix epoch until which to
+                find data points
+            resolution (optional[int]): resolution in milliseconds for data
+                returned, default=1000
+            timeout (optional[int]): how long to wait for response (in seconds)
+
+        Returns:
+            dictionary containing the data in the windows of the relevant time
+                series
+
+        """
+        logging.debug('Performing time series window elasticsearch for %s'
+                      % query)
+        url_to_get = '{0}/{1}?query={2}'.format(
+            self._api_endpoint, self._TIME_SERIES_WINDOW_ENDPOINT_SUFFIX,
+            query)
+        url_to_get += '&startMs=' + str(start_ms)
+        url_to_get += '&endMs=' + str(end_ms)
+        # for resolution, use API default (by leaving out of url)
+        if resolution is not None:
+            url_to_get += '&resolution=' + str(resolution)
+        _timeout = timeout or self._timeout
+        resp = self._get(url_to_get, session=self._api_session,
+                         timeout=_timeout)
+        resp.raise_for_status()
```

Comment:
take care of this incase of an exception. resp may not have these methods raise_for_status() and json()